syz-manager/manager.go

```diff
@@ -942,11 +943,14 @@ func (mgr *Manager) uploadReproAssets(repro *repro.Result) []dashapi.NewAsset {
 
 func (mgr *Manager) corpusInputHandler(updates <-chan corpus.NewItemEvent) {
 	for update := range updates {
-		if len(update.NewCover) != 0 && mgr.coverFilter != nil {
+		mgr.mu.Lock()
```

Comment:
Does this mutex fixing some existing bug? Or the rules for when coverFilter is initialized have changed in this PR?

Reply:
I think it must work fine without the mutex
1) The `corpusInputHandler` goroutine is started at the beginning, but it won't use `coverFilter` until the fuzzer has triaged some first corpus programs.
2) `httpCoverCover` relies on `checkDone`, which is set after `coverFilter` because `rpcserver.Server` calls `CoverageFilter` before `MachineChecked`.

It still looks brittle and not immediately obvious :(

For `.modules`, for example, we did have a real chance of a data race with `http.go`.

Reply:
Yes, modules are not well tested, and were are sloppy there.
Maybe all handlers should check check checkDone first. We probably could do it in middleware.

Reply:
Definitely not for all handlers -- e.g. it's better to still be able to render the main page even before the machine check is complete.

Reply:
Added a `checkDone` check to `modulesInfo()`.
PTAL.